### PR TITLE
Fix Classification Accuracy in Tests

### PR DESCRIPTION
**Pull request by abdel**
**Post-mortem**
The issue was kind of staring at us in the face, but easy to miss with the focus being solely on an implementation or a parameter issue.

The `predict` methods in Logistic Regression and Svm return the actual predicted label, so would be for example `1` or `0` in binary classification. We were getting the right predicted labels, but when outputting the accuracy to the screen we were instead calculating the mean error. The mean error would have been fine if we're actually looking at errors.

Since we were actually getting the predicted class, the `approxEquals` ended up not really working for this case. For example, if classTruth (1) and predicted (1), then `approxEquals` would indicate this is an error by returning `1` since `1-1 < EPS`.

With the SPECTF dataset, this lead to the `meanError` being ~0.95 on the test set but the classification accuracy was actually ~0.60, similar to scikit-learn and liblinear.

**Changes**
- Added a similar method to `FloatArray.approxEquals`, but instead checking if value from `arr1` is equal to the value from `arr2`, then return 1 otherwise return 0.
- Parameterised the "expected" train/test accuracy, so we can specify different assertions for each dataset
- Updated LogisticRegression & Svm tests to use the correct classification accuracy, and used the results from scikit-learn as the expected train/test accuracy for the assertions

**NOTE:** Let me know if the `FloatArray.equals` method should be moved elsewhere, or if we already have something similar that I may have missed.

### CODE REVIEW

**Comment by affogato** [comment]
as discussed, let's fix the `approxEquals()` bug, and remove the new `equals()` method.
